src/cascadia/TerminalCore/Terminal.cpp

```diff
@@ -411,6 +411,13 @@ try
     // before, and shouldn't be now either.
     _scrollOffset = originalOffsetWasZero ? 0 : static_cast<int>(::base::ClampSub(_mutableViewport.Top(), newVisibleTop));
 
+    // Make sure that the new cursor position is still in the mutable viewport
+    const auto currentNewPos = _mainBuffer->GetCursor().GetPosition();
+    if (currentNewPos.y < proposedTop)
+    {
+        _mainBuffer->GetCursor().SetPosition(til::point{ currentNewPos.x, proposedTop });
+    }

```

Comment:
I believe I once read somewhere in our repository that adjusting the viewport to fit the cursor is much better than adjusting the cursor. I'm not sure what the reason was back then, but I can see how a later write should always continue where a previous one left off.

Instead I would suggest moving this up to line 399 and do this (untested):
```cpp
proposedTop = std::min(proposedTop, newCursorPos.y);
```